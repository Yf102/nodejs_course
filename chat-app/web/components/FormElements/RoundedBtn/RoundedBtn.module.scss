.sendBtn {
  @apply rounded-full;
  background: #ffffffbd;
  width: 50px;
  height: 50px;
  line-height: 50px;
}