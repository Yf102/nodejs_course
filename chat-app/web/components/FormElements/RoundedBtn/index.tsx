import React, {
  ButtonHTMLAttributes,
  DetailedHTMLProps,
  forwardRef,
} from 'react'
import Image from 'next/image'
import cn from 'classnames'
import styles from './RoundedBtn.module.scss'

type BtnProps = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>

type BtnCustomProps = {
  src?: string
  alt?: string
  text?: string
  type?: 'button' | 'reset' | 'submit' | undefined
  className?: string
  onClick?: () => void
}

const RoundedBtn = forwardRef<HTMLButtonElement, BtnProps & BtnCustomProps>(
  (props, ref) => {
    const { className, src, alt, text, onClick, type = 'submit' } = props
    return (
      <button
        data-testid='rounded-btn-element'
        type={type}
        ref={ref}
        className={cn(className, styles.sendBtn)}
        onClick={onClick}
      >
        {src && alt && <Image src={src} alt={alt} width={50} height={50} />}
        {text}
      </button>
    )
  }
)

RoundedBtn.displayName = 'Button'

export default RoundedBtn
