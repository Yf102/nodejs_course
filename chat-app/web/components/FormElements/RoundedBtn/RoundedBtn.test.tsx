import React from 'react'
import { render, screen } from 'tests/test-utils'
import RoundedBtn from 'components/FormElements/RoundedBtn/index'

describe('RoundedBtn', () => {
  it('It renders RoundedBtn with image', async () => {
    const { asFragment } = render(
      <RoundedBtn type='submit' src='/icons/send_icon.png' alt='Send Button' />
    )
    expect(screen.getByTestId('rounded-btn-element')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  it('It renders RoundedBtn with text', async () => {
    const { asFragment } = render(<RoundedBtn type='button' text='Send' />)
    expect(screen.getByTestId('rounded-btn-element')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
})
