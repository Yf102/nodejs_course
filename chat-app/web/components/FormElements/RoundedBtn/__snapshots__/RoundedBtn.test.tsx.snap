// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedBtn It renders RoundedBtn with image 1`] = `
<DocumentFragment>
  <button
    class="sendBtn"
    data-testid="rounded-btn-element"
    type="submit"
  >
    <img
      alt="Send Button"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=%2Ficons%2Fsend_icon.png&w=128&q=75"
      srcset="/_next/image?url=%2Ficons%2Fsend_icon.png&w=64&q=75 1x, /_next/image?url=%2Ficons%2Fsend_icon.png&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
  </button>
</DocumentFragment>
`;

exports[`RoundedBtn It renders RoundedBtn with text 1`] = `
<DocumentFragment>
  <button
    class="sendBtn"
    data-testid="rounded-btn-element"
    type="button"
  >
    Send
  </button>
</DocumentFragment>
`;
